### PR TITLE
added zhop when not homing Z to prevent possible scratching after error while z probing (e.g. tap)

### PR DESCRIPTION
**Pull request by Surion79**
if X or Y is homed without z, it is always checked if a zhop is required. This helps if the probing was errored out, leaving the nozzle on the bed.

### CODE REVIEW

File: macros/base/homing/homing_overide.cfg [hunk 1]
```diff
@@ -77,7 +77,7 @@ gcode:
     {% endif %}
 
     G90
-
+
```
Comment (Frix-x) [inline macros/base/homing/homing_overide.cfg:80]: Be careful you added 4 spaces here. Can you remove them?

**Review by Frix-x — CHANGES_REQUESTED** [review]
Can you remove the spaces added, then I'm ok with this PR to be merged in the next version of Klippain :)

**Review by Frix-x — DISMISSED** [review]

**Comment by Surion79** [comment]
I added a new logic. while it also includes a fix. if Z is NOT requested but homed, it does not check for a required zhop. it just scratches.
The function has to be enabled in the variables. It does not check for previously homed axis. so all acis homing is lost after M84.

File: macros/base/homing/homing_overide.cfg [hunk 2]
```diff
@@ -94,6 +96,22 @@ gcode:
...
+                {% endif %}
+                G0 Z{homing_zhop} F{z_drop_speed}
+            {% endif %}
+        {% elif homing_force_zhop}
+            {% if verbose %}
+                { action_respond_info("Forcing z hop without homing z.") }
+            {% endif %}
+            SET_KINEMATIC_POSITION X=0 Y=0 Z=0
+            G0 Z{homing_zhop} F{z_drop_speed}
+            M84 # Motor off to "unhome" z for functional safety. This will also unhome all previously homed axis
```
Comment (Frix-x) [inline macros/base/homing/homing_overide.cfg:113]: And on this part, I had a look at the documentation and it's perfectly safe to do: `SET_KINEMATIC_POSITION Z=0` without impacting X and Y.
So this means that you can remove the variable and the use of M84. Just do a `SET_KINEMATIC POSITION Z=0` and then the Z hop :)

Also please change the respond_info to : "Homing X or Y while Z is not homed, forcing ZHOP"

File: user_templates/variables.cfg [hunk 3]
```diff
@@ -40,6 +40,10 @@ variable_force_homing_in_start_print: False
 ## Z hop before homing to avoid grinding
 ## the bed due to the gantry sag
 variable_homing_zhop: 5
+## Z hop before homing only X/Y without Z. In case while probing erroring out.
+## For functional safety reasons, if the following option is set to true, 
+## a M84 (Motors off) will be issued between forced Z hop and X/Y homing
+variable_homing_force_zhop: False
```
Comment (Frix-x) [inline user_templates/variables.cfg:46]: With the change asked before, I think this is not needed anymore :)

**Review by Frix-x — CHANGES_REQUESTED** [review]
Hi, sorry for the delay, I wanted to take some time to do it properly.
So I tried to analyse each case and see what would happens. Finally I think this can be made much more simpler without the need of a variable. See my comments :)

**Comment by Surion79** [comment]
@Frix-x I found and even better solution to solve everything. I didn't know that you can use M84 with axis parameter to disable only specific axis. So I changed M84 to M84 Z and did the changes you mentioned

**Comment by Surion79** [comment]
hmm, sad. i saw some links about stating that M84 <axis> works. seems not. need to check why it is documented but not working

File: macros/base/homing/homing_overide.cfg [hunk 4]
```diff
@@ -94,6 +93,13 @@ gcode:
             G0 Z{homing_zhop} F{z_drop_speed}
             {% set X, Y, Z = True, True, True %}
         {% endif %}
+    {% else %}
+        {% if verbose %}
+            { action_respond_info("Homing X or Y while Z is not homed, forcing ZHOP") }
+        {% endif %}
+        SET_KINEMATIC_POSITION Z=0
+        G0 Z{homing_zhop} F{z_drop_speed}
+        M84 Z
```
Comment (Frix-x) [inline macros/base/homing/homing_overide.cfg:102]: I think it's not needed as we know for sure that Z is already not homed

**Review by Frix-x — CHANGES_REQUESTED** [review]

**Comment by Surion79** [comment]
There are two topics in this PR, not just one

File: macros/base/homing/homing_overide.cfg (hunk 4 above)
Comment (Surion79) [reply macros/base/homing/homing_overide.cfg:102]: you described that setting the kinematic position would be like homing manually. this must be undone after the zhop

**Pull request closed without merging**